src/state.cc

```diff
@@ -68,9 +69,13 @@ Pool State::kDefaultPool("", 0);
 Pool State::kConsolePool("console", 1);
 const Rule State::kPhonyRule("phony");
 
-State::State() :
-  build_log_(new BuildLog()),
-  deps_log_(new DepsLog()) {
+State::State() : State(NULL) {}
```

Comment:
Allow a default constructed state?

Do you think people will construct the state by parts?

Reply:
Wouldn't State::State(Logger* logger = NULL) be sufficient?